Analysis/Tasks/PWGHF/HFDplusToPiKPiCandidateSelector.cxx

```diff
@@ -0,0 +1,295 @@
+                                            {0.2, 0.3, 0.3, 0.10, 5., 0.96, 0.000, 2.5},  /* 4<pt<5   */
+                                            {0.2, 0.3, 0.3, 0.10, 5., 0.96, 0.000, 2.5},  /* 5<pt<6   */
+                                            {0.2, 0.3, 0.3, 0.10, 5., 0.96, 0.000, 2.5},  /* 6<pt<7   */
+                                            {0.2, 0.3, 0.3, 0.10, 5., 0.96, 0.000, 2.5},  /* 7<pt<8   */
+                                            {0.2, 0.3, 0.3, 0.12, 5., 0.96, 0.000, 2.5},  /* 8<pt<10  */
+                                            {0.2, 0.3, 0.3, 0.12, 5., 0.96, 0.000, 2.5},  /* 10<pt<12 */
+                                            {0.2, 0.3, 0.3, 0.12, 5., 0.96, 0.000, 2.5},  /* 12<pt<16 */
+                                            {0.2, 0.3, 0.3, 0.12, 5., 0.96, 0.000, 2.5},  /* 16<pt<24 */
+                                            {0.2, 0.3, 0.3, 0.20, 5., 0.94, 0.000, 2.5}}; /* 24<pt<36 */
+
+/// Struct for applying Dplus to piKpi selection cuts
+struct HFDplusToPiKPiCandidateSelector {
+
+  Produces<aod::HFSelDplusToPiKPiCandidate> hfSelDplusToPiKPiCandidate;
+
+  Configurable<double> d_pTCandMin{"d_pTCandMin", 1., "Lower bound of candidate pT"};
+  Configurable<double> d_pTCandMax{"d_pTCandMax", 36., "Upper bound of candidate pT"};
+
+  Configurable<double> d_pidTPCMinpT{"d_pidTPCMinpT", 0.15, "Lower bound of track pT for TPC PID"};
+  Configurable<double> d_pidTPCMaxpT{"d_pidTPCMaxpT", 20., "Upper bound of track pT for TPC PID"};
+  Configurable<double> d_pidTOFMinpT{"d_pidTOFMinpT", 0.15, "Lower bound of track pT for TOF PID"};
+  Configurable<double> d_pidTOFMaxpT{"d_pidTOFMaxpT", 20., "Upper bound of track pT for TOF PID"};
+
+  Configurable<double> d_TPCNClsFindablePIDCut{"d_TPCNClsFindablePIDCut", 50., "Lower bound of TPC findable clusters for good PID"};
+  Configurable<bool> b_requireTPC{"b_requireTPC", true, "Flag to require a positive Number of found clusters in TPC"};
+  Configurable<double> d_nSigmaTPC{"d_nSigmaTPC", 3., "Nsigma cut on TPC"};
+  Configurable<double> d_nSigmaTOF{"d_nSigmaTOF", 3., "Nsigma cut on TOF"};
+
+  /// Gets corresponding pT bin from cut file array
+  /// \param candpT is the pT of the candidate
+  /// \return corresponding bin number of array
+  template <typename T>
+  int getpTBin(T candpT)
+  {
+    double pTBins[npTBins + 1] = {1., 2., 3., 4., 5., 6., 7., 8., 10., 12., 16., 24., 36.};
+    if (candpT < pTBins[0] || candpT >= pTBins[npTBins]) {
+      return -1;
+    }
+    for (int i = 0; i < npTBins; i++) {
+      if (candpT < pTBins[i + 1]) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  /// Selection on goodness of daughter tracks
+  /// \note should be applied at candidate selection
+  /// \param track is daughter track
+  /// \return true if track is good
+  template <typename T>
+  bool daughterSelection(const T& track)
+  {
+    if (track.sign() == 0) {
+      return false;
+    }
```

Comment:
If I see it correctly, the track charge sign is checked only here.
Since the track charge can never be 0, this condition is useless and can be removed.

Reply:
Removed. Although this is what is done in other selectors https://github.com/AliceO2Group/AliceO2/blob/610b9267036625b806e8182331876cab24cae66f/Analysis/Tasks/PWGHF/HFD0CandidateSelector.cxx#L97-L107 https://github.com/AliceO2Group/AliceO2/blob/610b9267036625b806e8182331876cab24cae66f/Analysis/Tasks/PWGHF/HFLcCandidateSelector.cxx#L83-L93
Is `daughterSelection(...)` needed at the moment? 

P.s I fixed also a problem related to #5566

Reply:
> Removed. Although this is what is done in other selectors

I know, the existing selectors also need some other fixes.

> Is `daughterSelection(...)` needed at the moment?

@nzardosh Any news about `tpcNClsFound`?